utils/api-scripts/src/fork-off.ts

```diff
@@ -0,0 +1,121 @@
+ * If you want to add any past module or part of a skipped module, add the prefix here manually.
+ *
+ * Any storage value’s hex can be logged via console.log(api.query.<module>.<call>.key([...opt params])),
+ * e.g. console.log(api.query.timestamp.now.key()).
+ *
+ * If you want a map/doublemap key prefix, you can do it via .keyPrefix(),
+ * e.g. console.log(api.query.system.account.keyPrefix()).
+ *
+ * For module hashing, do it via xxhashAsHex,
+ * e.g. console.log(xxhashAsHex('System', 128)).
+ */
+let prefixes = ['0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9' /* System.Account */];
+const skippedModulesPrefix = ['System', 'Session', 'Babe', 'Grandpa', 'GrandpaFinality', 'FinalityTracker', 'Authorship'];
+
+// Apparently not needed: To review
+// async function fixParachinStates(api: ApiPromise, chainSpec: any) {
+//     const skippedKeys = [
+//         api.query["parasScheduler"].sessionStartBlock.key()
+//     ];
+//     for (const k of skippedKeys) {
+//         delete chainSpec.genesis.raw.top[k];
+//     }
+// }
+
+async function main() {
+
+    // hexdump of runtime wasm binary, running it from the shell gives bad format error
+    execSync('cat ' + wasmPath + ' | hexdump -ve \'/1 "%02x"\' > ' + hexPath);
+
+    let api;
+    if (!fs.existsSync(schemaPath)) {
+        console.log(('Custom Schema missing, using default schema.'));
+        api = await ApiPromise.create({ provider });
+    } else {
+        const types = JSON.parse(fs.readFileSync(schemaPath, 'utf8'));
+        api = await ApiPromise.create({
+            provider,
+            types,
+        });
+    }
+
+    // storage.json is guaranteed to exists
+
+    let metadata = await api.rpc.state.getMetadata();
+    // Populate the prefixes array
+    let modules = metadata.asLatest.modules;
+    modules.forEach((module) => {
+        if (module.storage) {
+            if (!skippedModulesPrefix.includes(module.name.toString())) {
+                prefixes.push(xxhashAsHex(module.name.toString(), 128));
+            }
+        }
+    });
+
+    // blank starting chainspec guaranteed to exist
+
+    let storage: Storage = JSON.parse(fs.readFileSync(storagePath, 'utf8'));
+    let chainSpec = JSON.parse(fs.readFileSync(specPath, 'utf8'));
+
+    // Modify chain name and id
+    chainSpec.name = chainSpec.name + '-fork';
+    chainSpec.id = chainSpec.id + '-fork';
+    chainSpec.protocolId = chainSpec.protocolId;
+
+    // Grab the items to be moved, then iterate through and insert into storage
+    storage
+        .result
+        .filter((i) => prefixes.some((prefix) => i[0].startsWith(prefix)))
+        .forEach(([key, value]) => (chainSpec.genesis.raw.top[key] = value));
+
+    // Delete System.LastRuntimeUpgrade to ensure that the on_runtime_upgrade event is triggered
+    delete chainSpec.genesis.raw.top['0x26aa394eea5630e07c48ae0c9558cef7f9cce9c888469bb1a0dceaa129672ef8'];
```

Comment:
It would be better to show how these raw values are derived so this script can be generalized if we use it against a different chain.

Reply:
The raw values are derived from the Module name + storage element (+ eventual key in the map). Deriving the value I believe doesn't prevent the script from breaking
in case of a different chain, but it makes it easier to fix...